### PR TITLE
Screen refactor

### PR DESCRIPTION
**Pull request by NBCCF025**
## Description:

This PR refactors several screens to use common elements from Utils.kt. It primarily focuses on consolidating reusable UI components and optimizing layout structures across the affected screens. It aslo ensure that the tests pass.
The related tasks are #206 , #207 , #210 , #212 , #213 .

## Changes:
- Refactored multiple screens to use common UI components from `Utils.kt`, enhancing consistency and reducing code duplication.
- Updated the `StatisticsRow()` and the `InfoPopup()` in Utils.kt to be correctly displayed.
- Updated instances of `SquareButton()` in ProfileScreen.kt to ensure it fit entirely in the screen.
- Addapt the test tag and complete some test to pass.

## Testing:
- Updated and ran tests for all refactored screens to confirm they continue to function as expected.
- Performed UI checks to ensure that screen appearances and functionality remain unchanged for end users.

## Screen Concerned:
- ProfileScreen.kt
- SettingsScreen.kt
- FriendsListScreen.kt
- MyStatsScreen.kt
- ChallengeScreen.kt
- NewChallengeScreen.kt
- ChallengeHistoryScreen.kt

ChallengeHistoryScreen
ChallengeScreen
FriendsListScreen
MyStatsScreen
NewChallengeScreen
ProfileScreen
SettingsScreen

## Additional Notes:

This refactor does not introduce any functional changes. It focuses solely on improving code quality, readability, and maintainability by leveraging reusable components and streamlining screen layout structures. I also run into difficulties to modify CreateAChallenge so there are no change concerning this screen in this PR. Some element in the user model should be corrected as don't work as intended, for example the function concerning the `UserName`. Enhancement of the user model could be done next to avoid hardcoded value for the user information and statistic across all screens.

### CODE REVIEW

**Comment by NBCCF025** [comment]
The test coverage is less than 80%, that is due to the fact that there is no test for NewChallengeScreen.kt.
The test for this screen are done in Samy's branch.

**Comment by NBCCF025** [comment]
I reopen this PR to see if the CI pass, as I am trying to make another PR and can't pass the CI anymore.

**Comment by NBCCF025** [comment]
I put this PR as a draft as another one should be merged before. Now we also have to check why the CI doesn't pass anymore as I haven't done any change in this branch.

**Review by SaFouNaldo — COMMENTED** [review]
Nice work on cleaning up the challenges screen! Using the pre-built composable functions from Utils.kt really makes the code so much easier to read and follow. The refactoring is spot on, and it’s great to see how much simpler and more organized everything looks now. Keep it up—this kind of structure really makes a difference!

File: app/src/main/java/com/github/se/signify/ui/screens/challenge/NewChallengeScreen.kt [hunk 1]
```diff
@@ -55,111 +53,78 @@ fun NewChallengeScreen(
...
+        modifier =
+            Modifier.fillMaxWidth()
+                .border(2.dp, MaterialTheme.colorScheme.onPrimary)
+                .background(MaterialTheme.colorScheme.primary)
+                .padding(16.dp)
+                .testTag("OngoingChallengesBox")) {
+          Column(
+              horizontalAlignment = Alignment.CenterHorizontally,
+              modifier = Modifier.testTag("OngoingChallengesContent")) {
+                Text(
```
Comment (SaFouNaldo) [inline app/src/main/java/com/github/se/signify/ui/screens/challenge/NewChallengeScreen.kt:88]: UtilTextButton could have been a good choice here

File: app/src/main/java/com/github/se/signify/ui/screens/profile/SettingsScreen.kt [hunk 2]
```diff
@@ -88,135 +84,108 @@ fun SettingsScreen(
...
+          modifier =
+              Modifier.border(
+                      2.dp, MaterialTheme.colorScheme.outlineVariant, RoundedCornerShape(8.dp))
+                  .clip(RoundedCornerShape(8.dp))
+                  .background(MaterialTheme.colorScheme.background)
+                  .padding(horizontal = 24.dp, vertical = 16.dp),
+          horizontalArrangement = Arrangement.Center,
+          verticalAlignment = Alignment.CenterVertically,
+      ) {
+        Column {
```
Comment (SaFouNaldo) [inline app/src/main/java/com/github/se/signify/ui/screens/profile/SettingsScreen.kt:100]: As refactoring is the main purpose of the PR, you can add a new IconUtilis that does the same for the following icons

**Review by SaFouNaldo — APPROVED** [review]

File: app/src/main/java/com/github/se/signify/ui/screens/challenge/NewChallengeScreen.kt (hunk 1 above)
Comment (NBCCF025) [reply app/src/main/java/com/github/se/signify/ui/screens/challenge/NewChallengeScreen.kt:88]: This part is not a button, but the a text description for the box.

File: app/src/main/java/com/github/se/signify/ui/screens/profile/SettingsScreen.kt (hunk 2 above)
Comment (NBCCF025) [reply app/src/main/java/com/github/se/signify/ui/screens/profile/SettingsScreen.kt:100]: You are right, that could be a future improvement with more util functions, but I can't do it now. So I suggest that it becomes a task for a next sprint.

File: app/src/main/java/com/github/se/signify/ui/screens/challenge/NewChallengeScreen.kt (hunk 1 above)
Comment (SaFouNaldo) [reply app/src/main/java/com/github/se/signify/ui/screens/challenge/NewChallengeScreen.kt:88]: I meant utilText that I tought It was in Utils as it's something common to use.
In that case a feature work would be to add a utilTextBox for exmaple for more modularity

**Review by GRGool — APPROVED** [review]
Great work on this PR. AnnexScreenScaffold and MainScreenScaffold are used really well, and fit nicely.  StatisticsRow as well, in general its nice to be able to go the utils.kt and have the description of the composable. There's quite a bit of small padding/spacing adjustments, shows that you worked on the details for everything to look like before. Ran it locally as well as a sanity check, and it seems to be working and displayed correctly. I have left a comment about some @SuppressLint annotations that are I believe not useful anymore, as the padding argument isn't used now. Overall great job, I will approve.

File: app/src/main/java/com/github/se/signify/ui/screens/challenge/ChallengeHistoryScreen.kt [hunk 3]
```diff
@@ -1,31 +1,14 @@
...
-import androidx.compose.ui.unit.sp
-import com.github.se.signify.R
-import com.github.se.signify.ui.BackButton
-import com.github.se.signify.ui.UtilTextButton
+import com.github.se.signify.ui.AnnexScreenScaffold
+import com.github.se.signify.ui.NotImplementedYet
+import com.github.se.signify.ui.StatisticsRow
 import com.github.se.signify.ui.navigation.NavigationActions
 
 @SuppressLint("UnusedMaterialScaffoldPaddingParameter")
```
[the remark this answers is not in the document]
Comment (NBCCF025) [reply app/src/main/java/com/github/se/signify/ui/screens/challenge/ChallengeHistoryScreen.kt:14]: Good catch, it will be removed soon.

**Pull request merged**